test/sampler/test_povm_sampler_job.py

```diff
@@ -43,51 +47,116 @@ def test_initialization(self):
-        pm = generate_preset_pass_manager(optimization_level=2, backend=backend)
+        backend.set_options(seed_simulator=self.RNG_SEED)
+        pm = generate_preset_pass_manager(
+            optimization_level=2, backend=backend, seed_transpiler=self.RNG_SEED
+        )
 
         qc_isa = pm.run(qc)
 
-        measurement = ClassicalShadows(2, seed_rng=13)
-        runtime_sampler = RuntimeSampler(backend=backend)
+        measurement = ClassicalShadows(2, seed_rng=self.RNG_SEED)
+        runtime_sampler = RuntimeSampler(mode=backend)
         povm_sampler = POVMSampler(runtime_sampler)
         job = povm_sampler.run(pubs=[qc_isa], shots=128, povm=measurement)
-        job.save_metadata(filename="saved_metadata.pkl")
         tmp = job.base_job
 
-        job_recovered = POVMSamplerJob.recover_job(filename="saved_metadata.pkl", base_job=tmp)
-        self.assertIsInstance(job_recovered, POVMSamplerJob)
-        result = job_recovered.result()
-        pub_result = result[0]
-        observable = SparsePauliOp(["II", "XX", "YY", "ZZ"], coeffs=[1, 1, -1, 1])
-        post_processor = POVMPostProcessor(pub_result)
-        exp_value, std = post_processor.get_expectation_value(observable)
-        self.assertAlmostEqual(exp_value, 3.53125)
-        self.assertAlmostEqual(std, 0.3590672895231641)
+        with self.subTest("Save job with specific filename."):
+            job.save_metadata(filename="saved_metadata.pkl")
+            job_recovered = POVMSamplerJob.recover_job(filename="saved_metadata.pkl", base_job=tmp)
+            self.assertIsInstance(job_recovered, POVMSamplerJob)
+            result = job_recovered.result()
+            pub_result = result[0]
+            observable = SparsePauliOp(["II", "XX", "YY", "ZZ"], coeffs=[1, 1, -1, 1])
+            post_processor = POVMPostProcessor(pub_result)
+            exp_value, std = post_processor.get_expectation_value(observable)
+            self.assertAlmostEqual(exp_value, 4.445312500000001)
+            self.assertAlmostEqual(std, 0.3881881421165156)
+
+        with self.subTest("Save job with default filename."):
+            job.save_metadata()
+            job_recovered = POVMSamplerJob.recover_job(
+                filename=f"job_metadata_{job.base_job.job_id()}.pkl", base_job=tmp
+            )
+            self.assertIsInstance(job_recovered, POVMSamplerJob)
+            result = job_recovered.result()
+            pub_result = result[0]
+            observable = SparsePauliOp(["II", "XX", "YY", "ZZ"], coeffs=[1, -2, 1, 1])
+            post_processor = POVMPostProcessor(pub_result)
+            exp_value, std = post_processor.get_expectation_value(observable)
+            self.assertAlmostEqual(exp_value, -1.3906250000000009)
+            self.assertAlmostEqual(std, 0.6732583954195841)
+
+        with self.subTest("Test default ``base_job``."):
+            # TODO
+            # It requires QiskitRuntimeService. How can we test this ?
+            pass
```

Comment:
I don't have an idea from the top of my head.
Will think about this some more. For now, could you open an issue to remind us to get back to this?
I would assume that the [`qiskit-ibm-runtime`](https://github.com/Qiskit/qiskit-ibm-runtime) repo might have some tests of this which we could use for inspiration.

Reply:
Okay, I opened #43